crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -480,16 +525,25 @@ impl EventHandler {
                 // channel according to its
                 // `max_inbound_htlc_value_in_flight_percent_of_channel`
                 // configuration value
-                let temp_channel_id = self
+                let temp_channel_id = match self
                     .channel_manager
                     .create_channel(*target_node_id, channel_value, 0, 0, Some(user_config))
                     .map_err(|e| {
                         anyhow!(
                             "Could not create channel with {} due to {e:?}",
                             target_node_id
                         )
-                    })
-                    .expect("To open channel");
+                    }) {
+                    Ok(temp_channel_id) => temp_channel_id,
+                    Err(_e) => {
+                        // todo: check why this tracing fails here.
+                        // tracing::warn!("Failed to open just in time channel. Error: {e:?}");
```

Comment:
@luckysori: Do you know why this is failing? The event_handler panics at `backtrace::trace_unsynchronized`.

Reply:
commenting to follow this...

Reply:
Haha I've never come across this, I'll take a closer look.

Reply:
Could not reproduce in a test (where I caused a disconnect an saw the log being emitted).